internal/bgp/bgpfrr/config.go

```diff
@@ -0,0 +1,127 @@
+package bgpfrr
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"text/template"
+)
+
+// TODO configFileName will need to be set to the filename of the FRR configuration
+// file in the container.
+var configFileName = "frr.conf"
+
+const configTemplate = `
+log stdout {{.Loglevel}}
+hostname {{.Hostname}}
+
+{{range .Routers -}}
+router bgp {{.Asn}}
+  no bgp default ipv4-unicast
+  bgp router-id {{.RouterId}}
+{{range .Neighbors }}
+  neighbor {{.Addr}} remote-as {{.Asn}}
+{{- end }}
+{{range $n := .Neighbors -}}
+{{range .Advertisements }}
+  address-family {{.Version}} unicast
+    neighbor {{$n.Addr}} activate
+    network {{.Prefix}}
+  exit-address-family
+{{- end}}
+{{end}}
+{{end}}
+`
+
+type frrConfig struct {
+	Loglevel string
+	Hostname string
+	Routers  map[string]*routerConfig
+}
+
+type routerConfig struct {
+	Asn       uint32
+	RouterId  string
+	Neighbors map[string]*neighborConfig
+}
+
+type neighborConfig struct {
+	Asn            uint32
+	Addr           string
+	Advertisements map[string]*advertisementConfig
+}
+
+type advertisementConfig struct {
+	Version string
+	Prefix  string
+}
+
+// routerName() defines the format of the key of the "Routers" map in the
+// frrConfig struct.
+func routerName(srcAddr string, myAsn uint32) string {
+	return fmt.Sprintf("%d@%s", myAsn, srcAddr)
```

Comment:
you won't have more than 1 myASN/speaker

Reply:
I'll resolve this comment as it is a duplicate of the discussion above.